Content.Server/Nyanotrasen/Shipyard/Systems/ShipyardSystem.Consoles.cs

```diff
@@ -0,0 +1,240 @@
+using Content.Server.Popups;
+using Content.Server.Cargo.Systems;
+using Content.Server.Shipyard.Components;
+using Content.Server.Shuttles.Components;
+using Content.Server.Station.Systems;
+using Content.Shared.GameTicking;
+using Content.Shared.Shipyard.Events;
+using Content.Shared.Shipyard.BUI;
+using Content.Shared.Shipyard.Prototypes;
+using Content.Shared.Shipyard.Components;
+using Content.Server.Cargo.Components;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+using Robust.Shared.Players;
+using Robust.Shared.Prototypes;
+using Content.Shared.Access.Systems;
+using Content.Shared.Access.Components;
+using Content.Shared.Shipyard;
+
+namespace Content.Server.Shipyard.Systems
+{
+    public sealed class ShipyardConsoleSystem : SharedShipyardSystem
+    {
+        [Dependency] private readonly AccessReaderSystem _accessSystem = default!;
+        [Dependency] private readonly PopupSystem _popup = default!;
+        [Dependency] private readonly UserInterfaceSystem _uiSystem = default!;
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [Dependency] private readonly ShipyardSystem _shipyard = default!;
+        [Dependency] private readonly StationSystem _station = default!;
+        [Dependency] private readonly CargoSystem _cargo = default!;
+
+        public void InitializeConsole()
+        {
+            SubscribeLocalEvent<ShipyardConsoleComponent, ShipyardConsolePurchaseMessage>(OnPurchaseMessage);
+            //SubscribeLocalEvent<ShipyardConsoleComponent, ShipyardConsoleSellMessage>(OnSellMessage);
```

Comment:
So on my fork, this is player-centric and not station-centric, with the system storing a deed component onto the players personal bank account component.  The players are then able to sell their ship back to the shipyard.  I rewrote the purchasing part to use the stations bank account instead and removed the selling button from the UI, but since the selling code and logic was complete, I figured it would be better to leave the feature commented out if anyone in the future wanted to implement selling the ships back.

The back end code for selling is still implemented along with an admin command, so that admins are able to easily manipulate shuttles by "buying"/"selling" them in the console.  The sell command also Stull runs all the safety checks of isdocked to station and holds no organics.

Reply:
If you want the selling code that I commented out removed, I can do so.  When porting this over, I just thought that of any features this one might actually have a use, if not now then maybe sometime down the line.